monitor/monitor.go

```diff
@@ -145,14 +195,140 @@ func (m *Monitor) observeSTH() {
 	m.logger.Printf("STH for %q verified. Timestamp: %s Age: %s\n", m.logURI, ts, sthAge)
 }
 
+// randKey generates a random ECDSA private key or panics.
+func randKey() *ecdsa.PrivateKey {
+	key, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	if err != nil {
+		panic(fmt.Sprintf("Unable to generate random ECDSA key: %s\n", err.Error()))
+	}
+	return key
+}
+
+// randSerial generates a random *bigInt to use as a certificate serial or
+// panics.
+func randSerial() *big.Int {
+	serial, err := rand.Int(rand.Reader, big.NewInt(math.MaxInt64))
+	if err != nil {
+		panic(fmt.Sprintf("Unable to generate random certificate serial: %s\n", err.Error()))
+	}
+	return serial
+}
+
+// issueCertificate uses the monitor's certIssuer and certIssuerKey to generate
+// a leaf-certificate that can be submitted to a log. The certificate's subject
+// common name will be the monitor's subjCNPrefix and the first three bytes of
+// the random serial number encoded in hex.
+func (m *Monitor) issueCertificate() (*x509.Certificate, error) {
+	if m.certIssuerKey == nil {
+		return nil, errors.New("cannot issueCertificate with nil certIssuerKey")
+	}
+
+	certKey := randKey()
+	serial := randSerial()
+	template := &x509.Certificate{
+		Subject: pkix.Name{
+			CommonName: subjCNPrefix + hex.EncodeToString(serial.Bytes()[:3]),
```

Comment:
I'd like to emit certificates that generally pass cablint and similar things, to avoid making monitors who run linters produce special exclusions for us. I'm pretty sure CN for leaf certificates has to be a hostname. So let's pick a random nonexistent hostname under a domain we control, e.g. `<random>.woodpecker.testing.letsencrypt.org`.